File: BEFramework/BEFramework/Internal/Commands/REST/Command/BERESTSessionCommand.m
//
//  BERESTSessionCommand.m
//  BEFramework
//
//  Created by Loc Nguyen on 5/26/16.
//  Copyright © 2016 Loc Nguyen. All rights reserved.
//

#import "BERESTSessionCommand.h"
#import "BEHTTPRequest.h"

@implementation BERESTSessionCommand

+ (instancetype)getCurrentSessionCommandWithSessionToken:(nullable NSString *)sessionToken {
  return [self commandWithHTTPPath:@"sessions/me"
                        httpMethod:BEHTTPRequestMethodGET
                        parameters:nil
                      sessionToken:sessionToken];
}

@end

File: LoginWithSDK/csbm/src/main/java/com/csbm/BERESTSessionCommand.java
/*
 * Copyright (c) 2015-present, Parse, LLC.
 * All rights reserved.
 *
 * This source code is licensed under the BSD-style license found in the
 * LICENSE file in the root directory of this source tree. An additional grant
 * of patent rights can be found in the PATENTS file in the same directory.
 */
package com.csbm;


import com.csbm.http.BEHttpRequest;

import org.json.JSONObject;

/** package */ class BERESTSessionCommand extends BERESTCommand {

  public static BERESTSessionCommand getCurrentSessionCommand(String sessionToken) {
    return new BERESTSessionCommand(
        "sessions/me", BEHttpRequest.Method.GET, null, sessionToken);
  }

  public static BERESTSessionCommand revoke(String sessionToken) {
    return new BERESTSessionCommand(
        "logout", BEHttpRequest.Method.POST, new JSONObject(), sessionToken);
  }

  public static BERESTSessionCommand upgradeToRevocableSessionCommand(String sessionToken) {
    return new BERESTSessionCommand(
        "upgradeToRevocableSession", BEHttpRequest.Method.POST, new JSONObject(), sessionToken);
  }

  private BERESTSessionCommand(
      String httpPath,
      BEHttpRequest.Method httpMethod,
      JSONObject jsonParameters,
      String sessionToken) {
    super(httpPath, httpMethod, jsonParameters, sessionToken);
  }
}
